bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-notifications.js

```diff
@@ -89,12 +90,417 @@ export default function defineOHBlocks_Notifications (f7, isGraalJs) {
+    const actionButton2Type = (actionButton2 !== '') ? blockGetCheckedInputType(block, 'actionButton2') : undefined
+    let actionButton3 = javascriptGenerator.valueToCode(block, 'actionButton3', javascriptGenerator.ORDER_ATOMIC)
+    const actionButton3Type = (actionButton3 !== '') ? blockGetCheckedInputType(block, 'actionButton3') : undefined
+    let users = javascriptGenerator.valueToCode(block, 'usersInput', javascriptGenerator.ORDER_ATOMIC)
+
+    function getActionButtonCode (titleAndAction, buttonActionType) {
+      let actionButtonCode
+      if (buttonActionType === 'String') {
+        titleAndAction = titleAndAction.replace(/'/g, '').split('=')
+        if (titleAndAction.length === 2) {
+          let title = titleAndAction[0]
+          let buttonAction = titleAndAction[1].replace(/ /g, '')
+          actionButtonCode = `.addActionButton('${title}','${buttonAction}')`
+        }
+      } else {
+        actionButtonCode = titleAndAction
+      }
+      return actionButtonCode
+    }
+
+    let iconCode = (icon === '') ? '' : `.withIcon(${icon})`
+    let tagCode = (tag === '') ? '' : `.withTag(${tag})`
+    let headerTitleCode = (headerTitle === '') ? '' : `.withTitle(${headerTitle})`
+    let referenceCode = (reference === '') ? '' : `.withReferenceId(${reference})`
+    let mediaCode = (media === '') ? '' : `.withMediaAttachmentUrl(${media})`
+    let actionCode = (action === '') ? '' : (actionType === 'String') ? `.withOnClickAction(${action})` : `.withOnClickAction('${action}')`
+    let usersCode = (users === '') ? '' : `.addUserId(...${users}.split(','))`
+
+    let actionButton1Code = getActionButtonCode(actionButton1, actionButton1Type)
+    let actionButton2Code = getActionButtonCode(actionButton2, actionButton2Type)
+    let actionButton3Code = getActionButtonCode(actionButton3, actionButton3Type)
+    let logOnlyCode = (type === 'cloud') ? '' : '.logOnly()'
+
+    if (isGraalJs) {
+      return `actions.notificationBuilder(${message})${usersCode}${headerTitleCode}${iconCode}${tagCode}${referenceCode}${mediaCode}${actionCode}${actionButton1Code}${actionButton2Code}${actionButton3Code}${logOnlyCode}.send()\n`
+    } else {
+      throw new Error(unavailMsg)
+    }
+  }
+
+  Blockly.Blocks['oh_cloudNotificationButton'] = {
+    init: function () {
+      const actionButtonImageField = new Blockly.FieldImage(actionButtonImage, 15, 15, undefined)
+      this.appendValueInput('label')
+        .appendField(actionButtonImageField, 'actionButtonHint')
+        .appendField('label')
+        .setCheck('String')
+      this.appendValueInput('action')
+        .appendField('on-click action')
+        .setCheck(['String', 'oh_notificationAction'])
+      this.setInputsInline(false)
+      this.setColour(0)
+      this.setTooltip('Notification Button with label and action')
+      this.setOutput(true, 'oh_actionbutton')
```

Comment:
More concise naming:

```suggestion
      this.setOutput(true, 'oh_notificationActionButton')
```

Reply:
I don't like the type to be the same as the name of the button... I rather keep it as it is.

Reply:
`oh_actionbutton` IMO is a bit too generic ...

Reply:
I have changed it to something that is no block name.